plutus-ledger/src/Ledger/Index.hs

```diff
@@ -397,11 +350,15 @@ minAdaTxOut = Ada.lovelaceOf 2_000_000
 -- slightly more expensive than 1 Ada. So, to be on the safe side, we set the
 -- minimum Ada of each transaction output to 2 Ada.
 checkMinAdaInTxOutputs :: ValidationMonad m => Tx -> m ()
-checkMinAdaInTxOutputs t@Tx { txOutputs } =
-    for_ txOutputs $ \txOut ->
-        if Ada.fromValue (txOutValue txOut) >= minAdaTxOut
+checkMinAdaInTxOutputs t@Tx { txOutputs } = do
+    params <- vctxParams <$> ask
+    for_ txOutputs $ \txOut -> do
+        let
+            minAdaTxOut' = either (const minAdaTxOut) id $
```

Comment:
Mmm... I'm wondering whether we should keep this `minAdaTxOut`. What do you think @sjoerdvisscher ?

Reply:
I have started in that direction, to remove it everywhere, but it seems to be a next step as it breaks more tests and introduces more changes in the source tree (the PR is big already).

But yes, I would like to hear if it's a valid direction. I assume that we don't need it anymore.

Reply:
Yes, `minAdaTxOut` should no longer exist. It's fine to do that in a next PR, as long as you are confident we will be able to fix all tests.